### PR TITLE
Insertions

### LINKED ISSUE #86: Include Nucleotide Insertions
Store nucleotide insertions in the database and make them queryable.

Details:
- Probably create subtasks and link them here to split this into smaller bits.

### PR DESCRIPTION
**Pull request by Taepper**
Resolves #86 #87 #164

Most of the LoC are test data and test queries :)

### CODE REVIEW

File: include/silo/common/aa_symbols.h [hunk 1]
```diff
@@ -36,135 +39,33 @@ enum class AA_SYMBOL : char {
...
-   AA_SYMBOL::GAP, AA_SYMBOL::A, AA_SYMBOL::C, AA_SYMBOL::D, AA_SYMBOL::E,
-   AA_SYMBOL::F,   AA_SYMBOL::G, AA_SYMBOL::H, AA_SYMBOL::I, AA_SYMBOL::K,
-   AA_SYMBOL::L,   AA_SYMBOL::M, AA_SYMBOL::N, AA_SYMBOL::P, AA_SYMBOL::Q,
-   AA_SYMBOL::R,   AA_SYMBOL::S, AA_SYMBOL::T, AA_SYMBOL::V, AA_SYMBOL::W,
-   AA_SYMBOL::Y,   AA_SYMBOL::B, AA_SYMBOL::Z, AA_SYMBOL::X, AA_SYMBOL::STOP,
-};
+   static constexpr std::string_view symbol_name = "Amino Acid";
+   static constexpr std::string_view symbol_name_lower_case = "amino acid";
+   static constexpr std::string_view symbol_name_upper_case = "AMINO ACID";
+   static constexpr std::string_view symbol_name_short = "AA";
```
Comment (JonasKellerer) [inline include/silo/common/aa_symbols.h:50]: Please check case style of constexpr values in this file. They should be UPPER_CASE.

File: include/silo/common/nucleotide_symbols.h [hunk 2]
```diff
@@ -31,37 +33,46 @@ enum class NUCLEOTIDE_SYMBOL : char {
...
-   NUCLEOTIDE_SYMBOL::B,
-   NUCLEOTIDE_SYMBOL::D,
-   NUCLEOTIDE_SYMBOL::H,
-   NUCLEOTIDE_SYMBOL::V,
-   NUCLEOTIDE_SYMBOL::N,
-};
+   static constexpr std::string_view symbol_name = "Nucleotide";
+   static constexpr std::string_view symbol_name_lower_case = "nucleotide";
+   static constexpr std::string_view symbol_name_upper_case = "NUCLEOTIDE";
+   static constexpr std::string_view symbol_name_short = "NUC";
```
Comment (JonasKellerer) [inline include/silo/common/nucleotide_symbols.h:44]: Please check case style of constexpr values in this file. They should be UPPER_CASE.

File: include/silo/storage/aa_store.h [hunk 3]
```diff
@@ -44,7 +44,7 @@ class AAPosition {
    explicit AAPosition(AA_SYMBOL symbol);
    explicit AAPosition(std::optional<AA_SYMBOL> symbol);
 
-   AASymbolMap<roaring::Roaring> bitmaps;
+   SymbolMap<AA_SYMBOL, roaring::Roaring> bitmaps;
    std::optional<AA_SYMBOL> symbol_whose_bitmap_is_flipped = std::nullopt;
 
    std::optional<silo::AA_SYMBOL> flipMostNumerousBitmap(uint32_t sequence_count);
```
Comment (JonasKellerer) [inline include/silo/storage/aa_store.h:50]: Not shown here, but `computeSize`, `runOptimize` and `shrinkToFit` are not implemented.

File: src/silo/common/aa_symbols.cpp [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+std::optional<char> silo::Util<silo::AA_SYMBOL>::findIllegalChar(const std::string& sequence) {
+   for (const auto aa_char : sequence) {
+      auto symbol = Util<silo::AA_SYMBOL>::charToSymbol(aa_char);
+      if (symbol == std::nullopt) {
+         return aa_char;
+      }
+   }
+   return std::nullopt;
+}
```
Comment (JonasKellerer) [inline src/silo/common/aa_symbols.cpp:138]: Please add a test, similar to `nucleotide_symbol.test.cpp`

**Review by JonasKellerer — CHANGES_REQUESTED** [review]
Nice addition. Soon lapis v1 will be gone :)

Mostly smaller things. I'll also add an issue that will check the linter in the CI.

File: endToEndTests/test/invalidQueries/insertionsInvalidSequence.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+  "testCaseName": "The insertions action with an invalid sequence",
+  "query": {
+    "action": {
+      "type": "Insertions",
+      "column": "insertions",
+      "sequenceName": "S"
```
[the remark this answers is not in the document]
Comment (Taepper) [reply endToEndTests/test/invalidQueries/insertionsInvalidSequence.json:7]: This also shows that it does wrongly look in the amino acids for normal insertions for, but I can add it in an additional and rename this one

File: include/silo/common/symbols.h [hunk 6]
```diff
@@ -0,0 +1,13 @@
+#ifndef SILO_SYMBOLS_H
+#define SILO_SYMBOLS_H
+
+#include <cstdint>
+
+namespace silo {
+
+template <typename T>
+class Util {};
```
[the remark this answers is not in the document]
Comment (Taepper) [reply include/silo/common/symbols.h:9]: But this bloats all the Util<...> calls and repeats the word symbol

File: endToEndTests/package.json [hunk 7]
```diff
@@ -7,7 +7,7 @@
   "dependencies": {
     "chai": "^4.3.7",
     "mocha": "^10.2.0",
-    "prettier": "^3.0.2",
+    "prettier": "^3.0.1",
```
[the remark this answers is not in the document]
Comment (Taepper) [reply endToEndTests/package.json:10]: I just tried it to fix the CI

[the remark this answers is not in the document]
Comment (Taepper) [reply endToEndTests/package.json:10]: Will revert the commit before merging

File: include/silo/common/symbols.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (JonasKellerer) [reply include/silo/common/symbols.h:9]: Then its fine for me

**Review by JonasKellerer — APPROVED** [review]

**Pull request merged**